### PR TITLE
Adjust GC related settings for tests

### PR DESCRIPTION
**Pull request by arhimondr**
## Description

To exclude the possibility of GCLocker and PreventiveGC related issues

## Additional context and related issues

https://github.com/trinodb/trino/issues/15293

## Release notes

(X) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by findepi** [comment]
> To exclude the possibility of GCLocker and PreventiveGC related issues

please include this in commit message and/or code comment

File: pom.xml [hunk 1]
```diff
@@ -92,6 +92,9 @@
              Force bigger region size in attempt to help G1 utilize heap fully. -->
         <air.test.jvm.additional-arguments>
             -XX:G1HeapRegionSize=32M
+            -XX:+UnlockDiagnosticVMOptions
+            -XX:GCLockerRetryAllocationCount=10
```
Comment (findepi) [inline pom.xml:96]: only 10?

File: pom.xml [hunk 2]
```diff
@@ -92,6 +92,9 @@
              Force bigger region size in attempt to help G1 utilize heap fully. -->
         <air.test.jvm.additional-arguments>
             -XX:G1HeapRegionSize=32M
+            -XX:+UnlockDiagnosticVMOptions
```
Comment (findepi) [inline pom.xml:95]: is it required to set XX:GCLockerRetryAllocationCount ?

File: pom.xml (hunk 1 above)
Comment (arhimondr) [reply pom.xml:96]: It's 2 by default, I think 10 should be sufficient. Do you think it should be higher?

File: pom.xml (hunk 2 above)
Comment (arhimondr) [reply pom.xml:95]: It's actually needed for both :smile:

File: pom.xml (hunk 1 above)
Comment (findepi) [reply pom.xml:96]: oh, thought default is about 10. fine.
maybe add `<!-- bump from default of 2 -->` ?

**Review by findepi — APPROVED** [review]
% comments

**Pull request merged**